src/main/java/com/amazon/randomcutforest/serialize/SerializedRandomCutForest.java

```diff
@@ -0,0 +1,305 @@
+            this.point = point;
+        }
+
+        public double getWeight() {
+            return weight;
+        }
+
+        public void setWeight(double weight) {
+            this.weight = weight;
+        }
+
+        public long getSequenceIndex() {
+            return sequenceIndex;
+        }
+
+        public void setSequenceIndex(long sequenceIndex) {
+            this.sequenceIndex = sequenceIndex;
+        }
+    }
+
+    private static class Sampler {
+        private WeightedSamples[] weightedSamples;
+        private int sampleSize;
+        private double lambda;
+        private Random random;
+        private long entriesSeen;
+
+        public WeightedSamples[] getWeightedSamples() {
+            return weightedSamples;
+        }
+
+        public void setWeightedSamples(WeightedSamples[] weightedSamples) {
+            this.weightedSamples = weightedSamples;
+        }
+
+        public int getSampleSize() {
+            return sampleSize;
+        }
+
+        public void setSampleSize(int sampleSize) {
+            this.sampleSize = sampleSize;
+        }
+
+        public double getLambda() {
+            return lambda;
+        }
+
+        public void setLambda(double lambda) {
+            this.lambda = lambda;
+        }
+
+        public Random getRandom() {
+            return random;
+        }
+
+        public void setRandom(Random random) {
+            this.random = random;
+        }
+
+        public long getEntriesSeen() {
+            return entriesSeen;
+        }
+
+        public void setEntriesSeen(long entriesSeen) {
+            this.entriesSeen = entriesSeen;
+        }
+    }
+
+    private static class TreeUpdater {
+        public Sampler getSampler() {
+            return sampler;
+        }
+
+        public void setSampler(Sampler sampler) {
+            this.sampler = sampler;
+        }
+
+        public Tree getTree() {
+            return tree;
+        }
+
+        public void setTree(Tree tree) {
+            this.tree = tree;
+        }
+
+        private Sampler sampler;
+        private Tree tree;    
+    }
+
+    private static class Exec {
+        private TreeUpdater[] treeUpdaters;
+        private long totalUpdates;
+        private int threadPoolSize;
+
+        public TreeUpdater[] getTreeUpdaters() {
+            return treeUpdaters;
+        }
+
+        public void setTreeUpdaters(TreeUpdater[] treeUpdaters) {
+            this.treeUpdaters = treeUpdaters;
+        }
+
+        public long getTotalUpdates() {
+            return totalUpdates;
+        }
+
+        public void setTotalUpdates(long totalUpdates) {
+            this.totalUpdates = totalUpdates;
+        }
+
+        public int getThreadPoolSize() {
+            return threadPoolSize;
+        }
+
+        public void setThreadPoolSize(int threadPoolSize) {
+            this.threadPoolSize = threadPoolSize;
+        }
+    }
+
+    private static class Executor {
+        private String executor_type;
```

Comment:
Why is this in snake_case?

Reply:
That was an additional property added to the json result for denoting the type. Since it's part of json, I habitually used snake_case.